Paper.tex

```diff
@@ -2364,8 +2364,9 @@ \subsection{Instruction Set}
 &&&& $\boldsymbol{\mu}'_{\mathbf{s}}[0] \equiv \lVert \boldsymbol{\mu}_{\mathbf{o}} \rVert$ \\
 \midrule
 0x3e & {\small RETURNDATACOPY} & 3 & 0 & Copy output data from the previous call to memory. \\
+&&&& reverts if $\quad \boldsymbol{\mu}_{\mathbf{s}}[1] + \boldsymbol{\mu}_{\mathbf{s}}[2] > \lVert\boldsymbol{\mu}_{\mathbf{o}}\rVert$\\
 &&&& $\forall i \in \{ 0 \dots \boldsymbol{\mu}_{\mathbf{s}}[2] - 1\}: \boldsymbol{\mu}'_{\mathbf{m}}[\boldsymbol{\mu}_{\mathbf{s}}[0] + i ] \equiv
-\begin{cases} \boldsymbol{\mu}_{\mathbf{o}}[\boldsymbol{\mu}_{\mathbf{s}}[1] + i] & \text{if} \quad \boldsymbol{\mu}_{\mathbf{s}}[1] + i < \lVert \boldsymbol{\mu}_{\mathbf{o}} \rVert \\ 0 & \text{otherwise} \end{cases}$\\
+\boldsymbol{\mu}_{\mathbf{o}}[\boldsymbol{\mu}_{\mathbf{s}}[1] + i]$\\
```

Comment:
I would suggest to revert those changes.
The `RETURNDATACOPY` behavior is correct and don't need to be fixed.
The *Exceptional Halting* function (9.4.2.) already handle the case where we try to read past the returned data buffer.

![image](https://github.com/ethereum/yellowpaper/assets/5858657/3cf45af3-36ca-4510-a634-47a571e3baa8)

Reply:
I would still argue for altering the retiurndatacopy because it is a bit odd to include a case that will not happen,

what do you think about removing this line and keep the other alterations
![image](https://github.com/ethereum/yellowpaper/assets/28443115/59580e30-8444-47c0-a28a-b4cf886bc10d)

Reply:
The case where we read outside of the returned data buffer should never happen (because $Z$ is executed beforehand).
It was only described for completeness and that's why we should leave it as is imo. It also guarantee this property (exception should be caught beforehand)

![image](https://github.com/ethereum/yellowpaper/assets/5858657/33e02d4a-07ce-49cd-8a2f-3c108e528a6a)